modules/Bio/EnsEMBL/Gene.pm

```diff
@@ -1527,6 +1507,42 @@ sub havana_gene {
   return $ott;
 }
 
+=head2 biotype
+
+  Arg [1]    : Arg [1] : (optional) String - the biotype to set
+  Example    : my $biotype = $gene->biotype;
+               my $biotype = $gene->biotype('protin_coding');
+  Description: Returns the Biotype object of this gene.
+               When no biotype exists, defaults to 'protein_coding'.
+  Returntype : Bio::EnsEMBL::Biotype
+  Exceptions : none
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub biotype {
+  my ( $self, $new_value) = @_;
+
+  # was a term to set given?
+  if ( defined $new_value ) {
+    $self->{'biotype'} = $new_value;
+  }
+
+  # is there no biotype? default to 'protein_coding'
+  # this is legacy behaviour and should probably be revisited
+  if ( ! defined $self->{'biotype'}) {
+    $self->{'biotype'} = 'protein_coding';
+  }
+
+  my $biotype;
+  if( defined $self->adaptor() ) {
+    my $ba = $self->adaptor()->db()->get_BiotypeAdaptor();
+    $biotype = $ba->fetch_by_name_object_type( $self->{'biotype'}, 'gene' );
+  }
+
+  return ( $biotype || $self->{'biotype'} );
```

Comment:
This method should short-circuit to the data in the object, rather than querying the database whether it has a biotype property or not. It's a small but valuable efficiency tweak for users.

Reply:
Have modified this for efficiency.
if biotype() is never called, the biotype table is never queried.
When biotype() is first called, the Biotype object is retrieved and stored for future biotype() calls.